### PR TITLE
[WIP] Move some components to objects

### PR DESCRIPTION
**Pull request by laurarbunea**
I think some of these are more undecorated design patterns than components so therefore should probably live inside objects folder.

Not finished, but just want an opinion on where this is going

### CODE REVIEW

File: core/static_src/sass/objects/_objects.list.inline.scss [hunk 1]
```diff
@@ -1,3 +1,7 @@
+// ============================================================================
+// list inline 
+// ============================================================================
+
```
Comment (thibaudcolas) [inline core/static_src/sass/objects/_objects.list.inline.scss]: Please don't add headers like this in files that already have a header / documentation from the styleguide (SG) comment. There is no point in having this twice.

File: core/static_src/sass/objects/_objects.page.gutters.scss [hunk 2]
```diff
@@ -0,0 +1,43 @@
+// ============================================================================
+// Page gutters
+// ============================================================================
+//todo personally i think the comments should go at the bottom of the file
```
Comment (thibaudcolas) [inline core/static_src/sass/objects/_objects.page.gutters.scss]: The convention is to put documentation for a thing (function, class, file) at the top, otherwise it's more likely not to be seen and be useless / out of date.

File: docs/pattern-library/template.html [hunk 3]
```diff
@@ -76,7 +76,7 @@
         </div>
     </div>
 
-    <div class="theme--fred">
+    <div class="theme--dark">
```
Comment (thibaudcolas) [inline docs/pattern-library/template.html]: This isn't related to moving those components to objects so this should be a separate PR.

File: core/static_src/sass/tools/_tools.mixins.scss [hunk 4]
```diff
@@ -151,3 +151,13 @@
         object-fit: cover;
     }
 }
+
+@mixin border-dotted {
+    border-style: dotted;
+    border-color: $color-grey-light;
+    border-width: 0;
+
+    border-image-source: url('../images/dots-grey.svg');
```
Comment (thibaudcolas) [inline core/static_src/sass/tools/_tools.mixins.scss]: This seems very specific, does this belong in our pattern library?

File: core/static_src/sass/tools/_tools.mixins.scss [hunk 5]
```diff
@@ -151,3 +151,13 @@
...
+    border-image-slice: 23%;
+    border-image-repeat: round;
+}
```
Comment (thibaudcolas) [inline core/static_src/sass/tools/_tools.mixins.scss]: @laurarbunea please configure your text editor to always add new lines at ends of files (with the EditorConfig extension/plugin most likely).

File: core/static_src/sass/settings/_settings.global.scss [hunk 6]
```diff
@@ -36,14 +36,32 @@ $spacing-l: ($spacing + .5);      // 32px
...
 
-//Sections -- Vertical spacing
-$section-spacing:  3rem;
+// Spacing --- Grid
+// ----------------------------------------------------------------------------
+
+$spacing-grid-row: $spacing-s;
+$spacing-grid-row-s: $spacing-xs;
+
+$spacing-grid-col: $spacing;
```
Comment (thibaudcolas) [inline core/static_src/sass/settings/_settings.global.scss]: I think you should keep those variable naming changes outside of this PR – it makes it harder to consider changing the components to objects if there are other changes in there too.

File: core/static_src/sass/objects/_objects.list.border.scss [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+.list-border {
+    border-bottom: 1px solid $color-list-border;
+    border-left: 1px solid $color-list-border;
+	border-right: 1px solid $color-list-border;
+
+    li {
+		border-top: 1px solid $color-list-border;
+    }
+}
```
Comment (thibaudcolas) [inline core/static_src/sass/objects/_objects.list.border.scss]: This is all about looks so feels like this shouldn't be an object.

File: core/static_src/sass/objects/_objects.border.dotted.scss [hunk 8]
```diff
@@ -0,0 +1,41 @@
+// ============================================================================
+// Dotted Borders
+// ============================================================================
+
+$dotted-border-size: 2px;
+
+.dotted-border {
+    @include border-dotted;
+}
```
Comment (thibaudcolas) [inline core/static_src/sass/objects/_objects.border.dotted.scss]: This feels too specific to be an object, IMHO it should just be a mixin.

File: core/static_src/sass/objects/_objects.list.inline.scss (hunk 1 above)
Comment (laurarbunea) [inline core/static_src/sass/objects/_objects.list.inline.scss]: Sorry had these when the styleguide comments were at the bottom and yeah your totally right to have them at the top :) as it will hopefully keep the doc more up to date and stops me duplicating headers :)
